### ISSUE TITLE
Message isn't send if the yowsup-cli demos was triggered by a cron job

### ISSUE THREAD

**Issue by easterwood** [tgalal/yowsup#1122]
I have an odd behaviour on my system. If I send a message via the console, e.g. 
```
sudo /usr/local/bin/yowsup-cli demos -s 123456798 "Server is up" -c /mnt/data/projects/yowsup/whatsapp-config
```
then this message appears on my mobile phone. But if this is executed as a cron job (root) no message appears on the mobile phone. But the log says the message was sent.

Here is the debug log from the cron job version:
```
DEBUG:yowsup.stacks.yowstack:Initializing stack
DEBUG:yowsup.stacks.yowstack:Constructed Network Layer
DEBUG:yowsup.stacks.yowstack:Constructed Stanza Regulator Layer
DEBUG:yowsup.stacks.yowstack:Constructed Crypt Layer
DEBUG:yowsup.stacks.yowstack:Constructed Coder Layer
DEBUG:yowsup.stacks.yowstack:Constructed Logger Layer
DEBUG:yowsup.stacks.yowstack:Constructed Axolotl Layer
WARNING:yowsup.stacks.yowstack:Implicit declaration of parallel layers in a tuple is deprecated, pass a YowParallelLayer instead
DEBUG:yowsup.stacks.yowstack:Constructed Authentication Layer - Messages Layer - Receipt Layer - Ack Layer
DEBUG:yowsup.stacks.yowstack:Constructed Interface Layer
DEBUG:yowsup.layers.network.layer:Connecting to e7.whatsapp.net:443
DEBUG:yowsup.layers.logger.layer:tx:
<stream:features>
<readreceipts>
</readreceipts>
<groups_v2>
</groups_v2>
<privacy>
</privacy>
<presence>
</presence>
</stream:features>

DEBUG:yowsup.layers.logger.layer:tx:
<auth passive="true" user="49xxxxxxxx" mechanism="WAUTH-2">
p��� qn��f�Gt�?��Zص��!S��.�����q�֒f{�[�
HEX:7098a2b9031120716efbbe66f1104774f53f89c85ad8b51183c02153b4f82e83a69ea1ef0771e2d692667bb75b1e89
</auth>

DEBUG:yowsup.layers.logger.layer:rx:
<stream:features>
</stream:features>

DEBUG:yowsup.layers.logger.layer:rx:
<challenge>
�PH?Q�
       �;��+�DL
HEX:1ea250483f51fd0c1a1213df3bcaca2bed444c01
</challenge>

DEBUG:yowsup.layers.logger.layer:tx:
<response xmlns="urn:ietf:params:xml:ns:xmpp-sasl">
P��	��.t*���j��#S]qA(���ٲ���
3�{,
HEX:50121b90b809b4f415f6b11b2e742a8317cbe06a10d9dd23535d714128f016f806a9d9b2ad1ae0dd1d0a338d027b2c
</response>

DEBUG:yowsup.layers.logger.layer:rx:
<success status="active" kind="free" creation="1445370238" expiration="1476906238" props="4" t="1446580862">
�
 �CI�"m��5�nFA7_7w�
HEX:d30cba4349d0226de4c435f16e4641375f3777bd
</success>

DEBUG:yowsup.layers.logger.layer:tx:
<message to="49xxxxxx@s.whatsapp.net" type="text" id="1446580862-1">
<enc type="pkmsg" v="1">
���!H���|���>�����y_�+KQ�.WJ^<�up!���V<�U�\���*�FrW���7�{d�/"b3
!�&)��/�!*�|2�UFʲ~�%]ؠ�$�=bʖ="0q���>�Ec~�cT\u�Ԋݿ����#��3^6J�=�?#<�(�f	����t
0                                                                                     (���
HEX:330880fbf5031221050f48e8c3ec7c90a38c3ee58dfcb0d9795f1913fd2b4b51b32e574a5e3cf975701a2105109e7ff8a1563c061bdf55e65ce1fe822ae14672571ca0d71cd1378c7b64c82f2262330a2105a82629d7e12fe0212aa37c3204e95546cab27e81255dd8a0db24d83d62ca963d10171800223071ade2bd3ede45637e8e268963545c7512e4d48addbf90bfb59023aadc330f5e364aa33d8c3f233ceb28f666091cbea8078eb09bb6740f0b28eee1d4e80d3000
</enc>
</message>

DEBUG:yowsup.layers.logger.layer:rx:
<ack t="1446580862" from="49xxxxxxx@s.whatsapp.net" class="message" id="1446580862-1">
</ack>

INFO:yowsup.demos.sendclient.layer:Message sent

Yowsdown
```

**Comment by jlguardi** [comment]
Probably cron doens't use the same axolot.db so message is received by server but discarted by recipient.

**Comment by easterwood** [comment]
How can I determine which axolot.db is used by the cron job?

**Comment by jlguardi** [comment]
It depends on user which use cron. i.e: if you test the app with your user (MY_NAME) the axolot.db is located in /home/MY_NAME/.yowsup/PHONE/axolotl.db however, if you have programmed cron using root, it will use /root/.yowsup/PHONE/axolotl.db.

Try to configure cron from your user.

**Comment by easterwood** [comment]
I checked the example from above and I forgot to mention that the console test was executed with ```sudo```. 

Besides I can't test it with a normal user because only root has the rights to execute yowsup-cli. I think the reason for this is that it was build with root but this isn't the subject of this issue :)

**Comment by easterwood** [comment]
I thought about which user home was used by yowsup and I tried the command as user ```root``` and not via ```sudo```. The result was no message received by the mobile phone. So I deleted the ```axolotl.db``` file in the root home. After re-generation of the ```axolotl.db``` with user ```root``` I again tried to send a message as ```root``` and bingo. The message appeared on the mobile phone. Now it also works via a the root cron job. It seems that ```sudo ....``` user the current user home directory instead of root home.

Thx @jlguardi for your advise.
